src/Libraries/CoreNodeModelsWpf/Charts/Utilities/Export.cs

```diff
@@ -4,37 +4,62 @@
+        public static void ToPng(Chart control, GeoMap gcontrol = null)
         {
-            RenderTargetBitmap rtb = new RenderTargetBitmap((int)control.ActualWidth, (int)control.ActualHeight, 96, 96, PixelFormats.Pbgra32);
-            rtb.Render(control);
+            if (control == null && gcontrol == null) return;
 
-            PngBitmapEncoder png = new PngBitmapEncoder();
-            png.Frames.Add(BitmapFrame.Create(rtb));
-            MemoryStream stream = new MemoryStream();
-            png.Save(stream);
-            var image = System.Drawing.Image.FromStream(stream);
+            int img_width = control == null ? (int)gcontrol.ActualWidth : (int)control.ActualWidth;
+            int img_height = control == null ? (int)gcontrol.ActualHeight : (int)control.ActualHeight;
+            InMemorySkiaSharpChart skChart;
+            try
+            {
+                switch (control)
+                {
+                    case CartesianChart cartesianChart:
+                        skChart = new SKCartesianChart(cartesianChart) { Width = img_width, Height = img_height, };
+                        break;
+                    case PieChart pieChart:
+                        skChart = new SKPieChart(pieChart) { Width = img_width, Height = img_height, };
+                        break;
+                    default:
+                        skChart = new SKGeoMap(gcontrol) { Width = img_width, Height = img_height, };
+                        break;
+                }
 
-            SaveFileDialog dialog = new SaveFileDialog();
-            dialog.FileName = "NodeModelChart"; // Default file name
-            dialog.DefaultExt = ".png"; // Default file extension
-            dialog.Filter = "Image files (*.png) | *.png"; // Filter files by extension
+                SaveFileDialog dialog = new SaveFileDialog();
+                dialog.FileName = "NodeModelChart"; // Default file name
+                dialog.DefaultExt = ".png"; // Default file extension
+                dialog.Filter = "Image files (*.png) | *.png"; // Filter files by extension
 
-            // Show save file dialog box
-            Nullable<bool> result = dialog.ShowDialog();
+                // Show save file dialog box
+                bool? result = dialog.ShowDialog();
 
-            // Process save file dialog box results
-            if (result == true)
+                // Process save file dialog box results
+                if (result == true)
+                {
+                    // Save document
+                    string filename = dialog.FileName;
+                    skChart.SaveImage(filename, SkiaSharp.SKEncodedImageFormat.Png);
+                }
+            }
+            catch (Exception ex)
             {
-                // Save document
-                string filename = dialog.FileName;
-                image.Save(filename, ImageFormat.Png);
+                Console.WriteLine("Chart Image Export Error:" + ex.Message);
```

Comment:
Do we need this try-catch or was this also for your debugging?

Reply:
Is Console.WriteLine visible in the Dynamo console UI ? or at least in the console logs ?
Do we have access to the Dynamo Logger here ?
I say we should let the exception bubble up, unless it is something very specific you are trying to catch here @zeusongit .

Reply:
This was for test as well, LC2 has summed up all charts into 2 types, SKCartesian and Pie, but there is one more Geo charts which are not yet implemented, so was testing if any of the existing charts throw an exception while exporting image.